### ISSUE TITLE
Enum constant parameter flagged for MagicNumber

### ISSUE THREAD

**Issue by benwicks** [arturbosch/detekt#455]
I have an enum class with an integer as a constructor parameter. One of the enum constants calls the constructor with a number that violates the `MagicNumber` rule (e.g. `3`).

I do not want to disable the `MagicNumber` rule for the whole project. How can I prevent this check from failing on this file and do so at the smallest scope possible?

I am interested in contributing but would need some direction. Thank you!

**Comment by schalkms** [comment]
Thanks for reporting!
I agree. Detekt should not detect this case. 
You can of course supress it for now.

**Comment by arturbosch** [comment]
Have you read about the baseline.xml file where you can add false positives? (See readme)

**Comment by benwicks** [comment]
I did not know about the baseline. That could be useful! Thank you!

I see your point. The way I write the enums in question would be closer to:
```kotlin
enum class Bag(id: Int) {
    SMALL(1),
    EXTRA_LARGE(5)
}
```
So the enum constant name makes it clear what the "magic number" represents.

Is my best way to proceed to accept the `MagicNumber` rule violation and add it to my baseline?

**Comment by arturbosch** [comment]
Ok, in your case I would also say no MagicNumber as `id` says everything we need to know! 

We have no symbol resolution to get to the right constructor and check then if the name of the variable `is enought` (configurable in the config?) but there is a PR open which fixes named arguments as MagicNumbers, so you could write:

```kotlin
enum class Bag(id: Int) {
	SMALL(id = 1),
	EXTRA_LARGE(id = 5)
}
```

which is kinda dumb as you are in the class itself ...

**Comment by benwicks** [comment]
I see. I like the idea to configure no checking for enum classes.

Do you think it could be configurable down to the level of not checking just enum constant constructors? Or is that too fine-grained?

**Comment by arturbosch** [comment]
Should be doable, will look at this 👍

**Comment by Mauin** [comment]
What about a case like this? Are the numbers in this case more or less magic than in the examples above?

```
enum class Bag(id: Int, manufacturer: String, name: String, whatever: String) {
	SMALL(1, "brand", "smallbag", "something"),
	EXTRA_LARGE(5, "brand", "xlbag", "something")
}
```

**Comment by benwicks** [comment]
@Mauin I would want a way for that example to _not_ be flagged for `MagicNumber`.
